eventbrite/britecharts
# Adding support for negative area charts / allow scaling of y-axis below 0

Added support for _negative_ area charts (meaning that the stacked area reaches below y-values of 0) - in general the solution is really similar to the solution which was implemented for the line-chart as well some weeks ago. Highlighting the 0-axis in case of negative values is implemented as well, exactly as it was done by @cherurg in the line-chart feature to be consistent.

To avoid confusion:
In this context I speak of a _negative_ area chart, if the stacked-area is below 0. 

## Motivation and Context
This feature is missing until now, there is need for it - but currently there is no issue for it.

## How Has This Been Tested?

- Check if the first tick is negative in case of a _negative_ area chart
- Check that there exists the `horizontal-grid-line--highlighted` class in case of a _negative_ area chart
- Check that this class is _not_ there, in case of a _non-negative_ area chart

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/16606530/79036478-e098b000-7bc8-11ea-91bc-4ccf2dd69dea.png)

## Types of changes
- [ ] Refactor (changes the way we code something without changing its functionality)

## Checklist:
- [x] Latest master code has been merged into this branch
- [x] No commented out code (if required, place // TODO above with explanation)
- [ ] Code follows the [API Guidelines](http://eventbrite.github.io/britecharts/topics-index.html#toc5__anchor)
- [x] Added tests to cover changes
